src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -179,15 +179,19 @@ private RowMapBuffer getTransactionRows(BinlogConnectorEvent beginEvent) throws
 					Table table = tableCache.getTable(event.getTableID());
 
 					if ( table != null && shouldOutputEvent(table.getDatabase(), table.getName(), filter) ) {
-						for ( RowMap r : event.jsonMaps(table, lastHeartbeatPosition) )
+						for ( RowMap r : event.jsonMaps(table, lastHeartbeatPosition, temporaryQuery) )
 							buffer.add(r);
 					}
-
+					temporaryQuery = "";
 					break;
 				case TABLE_MAP:
 					TableMapEventData data = event.tableMapData();
 					tableCache.processEvent(getSchema(), this.filter, data.getTableId(), data.getDatabase(), data.getTable());
 					break;
+				case ROWS_QUERY:
+					RowsQueryEventData rqed = event.getEvent().getData();
+					temporaryQuery = rqed.getQuery();
+					break;
```

Comment:
It looks like this code is in a loop which runs for every update in a transaction. If there are multiple queries in a transaction, does that mean we only end with the last query?

Reply:
I suppose you mean the situation

<pre>
START TRANSACTION;
UPDATE `test` SET `field`='foo' WHERE `id`='1';
UPDATE `test` SET `field`='bar' WHERE `id`='1';
COMMIT;
</pre>

and in this case, there are two separate event in the binlog, not one.